### ISSUE TITLE
Fails to compile

### ISSUE THREAD

**Issue by DigiAngel** [radare/radare2#6595]
As of this morning:
~~~
In file included from line.c:40:0:
dietline.c: In function ‘r_line_readline_cb’:
dietline.c:1468:1: error: version control conflict marker in file
 <<<<<<< HEAD
 ^~~~~~~
dietline.c:1479:2: error: #else after #else
 #else
  ^~~~
dietline.c:1467:0: error: the conditional began here
 #if USE_UTF8

CC 2048.c
CC bp_arm.c
/opt/radare2/libr/../global.mk:41: recipe for target 'line.o' failed
make[3]: *** [line.o] Error 1
make[3]: *** Waiting for unfinished jobs....
CC magic.c
CC bp_bf.c
CC bp_mips.c
CC bp_ppc.c
CC softmagic.c
CC bp_x86.c
LD libr_search.so
Makefile:91: recipe for target 'cons' failed
make[2]: *** [cons] Error 2
make[2]: *** Waiting for unfinished jobs....
LD libr_bp.so
LD libr_magic.so
Makefile:28: recipe for target 'all' failed
make[1]: *** [all] Error 2
Makefile:47: recipe for target 'all' failed
make: *** [all] Error 2
~~~

Git info:
~~~
commit 53062962d6472e0a15f741b69fd642538420f3da
Author: Álvaro Felipe Melchor <alvaro.felipe91@gmail.com>
Date:   Wed Jan 25 21:58:00 2017 +0100

    Fix #6579 - mostly done by @leberus

diff --git a/libr/cons/dietline.c b/libr/cons/dietline.c
index f92295c46..547add5fc 100644
--- a/libr/cons/dietline.c
+++ b/libr/cons/dietline.c
@@ -1231,7 +1231,7 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
                                }
                                break;
                        default:
-                               buf[1] = r_line_readchar();
+                               buf[1] = r_line_readchar ();
                                if (buf[1] == -1) {
                                        r_cons_break_pop ();
                                        return NULL;
@@ -1465,9 +1465,14 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
 #endif
                        }
 #if USE_UTF8
-                       I.buffer.index += utflen;
+                       if ((I.buffer.index + utflen) < I.buffer.length &&
+                           I.buffer.index + utflen > utflen) {
+                               I.buffer.index += utflen;
+                       }
 #else
-                       I.buffer.index++;
+                       if ((I.buffer.index + 1) < I.buffer.length) {
+                               I.buffer.index++;
+                       }
 #endif
                        break;
                }
~~~

**Comment by leberus** [comment]
Which version are you using? That commit was reverted

**Comment by DigiAngel** [comment]
Not sure to be honest...I just did the sys/install.sh dance this morning and that's what I got.  However after nuking the /opt/radare dir and re-cloning all seems well:
~~~
commit 800379c475ba84a35c05633ca4806036975a4b23
Author: leberus <osalvador.vilardaga@gmail.com>
Date:   Fri Jan 27 10:06:31 2017 +0100

    Try to fix #6579 (#6589)

diff --git a/libr/cons/dietline.c b/libr/cons/dietline.c
index f92295c46..af2a9df0d 100644
--- a/libr/cons/dietline.c
+++ b/libr/cons/dietline.c
@@ -1465,9 +1465,13 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
 #endif
                        }
 #if USE_UTF8
-                       I.buffer.index += utflen;
+                       if ((I.buffer.index + utflen) <= I.buffer.length) {
+                                       I.buffer.index += utflen;
+                       }
 #else
-                       I.buffer.index++;
+                       if (I.buffer.index < I.buffer.length) {
+                               I.buffer.index++;
+                       }
 #endif
                        break;
                }
~~~
Thanks and sorry for the noise.

**Comment by leberus** [comment]
np, have fun ;)!
